manager/allocator/network.go

```diff
@@ -315,7 +315,7 @@ func (a *Allocator) doNetworkAlloc(ctx context.Context, ev events.Event) {
 	case state.EventUpdateService:
 		s := v.Service.Copy()
 
-		if !serviceAllocationNeeded(s, nc) {
+		if nc.nwkAllocator.IsServiceAllocated(s) {
```

Comment:
Yes as part of a bigger refactor I moved all the logic inside `serviceAllocationNeeded` into `nwkAllocator.IsServiceAllocated` which was needed for other reasons and as part of that made sure that the fix in #1301 is preserved in `IsServiceAllocated`.